samples/wifi/radio_test/src/nrf_wifi_radio_ficr_shell.c

```diff
@@ -106,75 +106,115 @@ static int nrf_wifi_radio_test_otp_read_params(const struct shell *shell,
-		(uint8_t)(val[MAC0_ADDR] >> 16), (uint8_t)(val[MAC0_ADDR] >> 24),
-		(uint8_t)(val[MAC0_ADDR + 1]), (uint8_t)(val[MAC0_ADDR + 1] >> 8));
-	shell_fprintf(shell, SHELL_INFO, "\n");
+	if (val[REGION_PROTECT + 0] != OTP_FRESH_FROM_FAB) {
+		shell_fprintf(shell, SHELL_INFO, "MAC0.ADDRESS0 = 0x%08x\n", val[MAC0_ADDR]);
+		shell_fprintf(shell, SHELL_INFO, "MAC0.ADDRESS1 = 0x%08x\n", val[MAC0_ADDR + 1]);
+		shell_fprintf(shell, SHELL_INFO, "MAC0.ADDRESS = %02x:%02x:%02x:%02x:%02x:%02x\n",
+			(uint8_t)(val[MAC0_ADDR]), (uint8_t)(val[MAC0_ADDR] >> 8),
+			(uint8_t)(val[MAC0_ADDR] >> 16), (uint8_t)(val[MAC0_ADDR] >> 24),
+			(uint8_t)(val[MAC0_ADDR + 1]), (uint8_t)(val[MAC0_ADDR + 1] >> 8));
+		shell_fprintf(shell, SHELL_INFO, "\n");
+
+		shell_fprintf(shell, SHELL_INFO, "MAC1.ADDRESS0 = 0x%08x\n", val[MAC1_ADDR]);
+		shell_fprintf(shell, SHELL_INFO, "MAC1.ADDRESS1 = 0x%08x\n", val[MAC1_ADDR + 1]);
+		shell_fprintf(shell, SHELL_INFO, "MAC1.ADDRESS = %02x:%02x:%02x:%02x:%02x:%02x\n",
+			(uint8_t)(val[MAC1_ADDR]), (uint8_t)(val[MAC1_ADDR] >> 8),
+			(uint8_t)(val[MAC1_ADDR] >> 16), (uint8_t)(val[MAC1_ADDR] >> 24),
+			(uint8_t)(val[MAC1_ADDR + 1]), (uint8_t)(val[MAC1_ADDR + 1] >> 8));
+		shell_fprintf(shell, SHELL_INFO, "\n");
+
+		shell_fprintf(shell, SHELL_INFO, "CALIB.XO = 0x%02x\n", val[CALIB_XO] & 0xFF);
+
+		shell_fprintf(shell, SHELL_INFO, "REGION_DEFAULTS = 0x%08x\n",
+			val[REGION_DEFAULTS]);
+		shell_fprintf(shell, SHELL_INFO, "\n");
+	}
+	return 0;
+}
 
-	shell_fprintf(shell, SHELL_INFO, "MAC1 : Reg0 = 0x%08x\n", val[MAC1_ADDR]);
-	shell_fprintf(shell, SHELL_INFO, "MAC1 : Reg1 = 0x%08x\n", val[MAC1_ADDR + 1]);
-	shell_fprintf(shell, SHELL_INFO, "MAC1 Addr   = %02x:%02x:%02x:%02x:%02x:%02x\n",
-		(uint8_t)(val[MAC1_ADDR]), (uint8_t)(val[MAC1_ADDR] >> 8),
-		(uint8_t)(val[MAC1_ADDR] >> 16), (uint8_t)(val[MAC1_ADDR] >> 24),
-		(uint8_t)(val[MAC1_ADDR + 1]), (uint8_t)(val[MAC1_ADDR + 1] >> 8));
-	shell_fprintf(shell, SHELL_INFO, "\n");
-	shell_fprintf(shell, SHELL_INFO, "CALIB_XO = 0x%02x\n", val[CALIB_XO] & 0xFF);
+static int nrf_wifi_radio_test_otp_read_retrim_version(const struct shell *shell,
+					size_t argc,
+					char  *argv[])
+{
+	unsigned int val[1];
+	unsigned int err;
 
+	err = read_otp_memory(REGION_PROTECT, &val[0], 1);
 
-	shell_fprintf(shell, SHELL_INFO, "REGION_DEFAULTS = 0x%08x\n",
-		val[REGION_DEFAULTS]);
-	shell_fprintf(shell, SHELL_INFO, "\n");
+	if (val[0] == OTP_FRESH_FROM_FAB) {
+		shell_fprintf(shell, SHELL_ERROR,
+				"Region is unprogrammed - program to enable R/W\n");
+	} else {
+		err = read_otp_memory(PRODRETEST_PROGVERSION, &val[0], 1);
+		shell_fprintf(shell, SHELL_INFO, "\nPRODRETEST.PROGVERSION = 0x%08x\n",
+		val[0]);
+		shell_fprintf(shell, SHELL_INFO, "\n");
+	}
+	return 0;
+}
+
+static int nrf_wifi_radio_test_otp_read_retrim_params(const struct shell *shell,
+					size_t argc,
+					char  *argv[])
+{
+	unsigned int val[RETRIM_LEN];
+	unsigned int err;
+
+	err = read_otp_memory(REGION_PROTECT, &val[0], 1);
```

Comment:
Can just pass `val` instead of `&val[0]`

Reply:
_read_otp_memory_ indexes val assuming it as an array.